packages/x-charts/src/internals/useAnimate.ts

```diff
@@ -0,0 +1,91 @@
+'use client';
+import * as React from 'react';
+import { ANIMATION_DURATION_MS, ANIMATION_TIMING_FUNCTION_JS } from './animation';
+import { Transition } from './Transition';
+import { shallowEqual } from './shallowEqual';
+import { useIsomorphicLayoutEffect } from './useIsomorphicLayoutEffect';
+
+export function useAnimate<Props extends {}, Elem extends Element>(
+  props: Props,
+  {
+    createInterpolator,
+    applyProps,
+    skip,
+    initialProps,
+  }: {
+    createInterpolator: (lastProps: Props, newProps: Props) => (t: number) => Props;
+    applyProps: (element: Elem, props: Props) => void;
+    skip?: boolean;
+    initialProps?: Props;
+  },
+) {
+  const lastInterpolatedPropsRef = React.useRef(initialProps ?? props);
```

Comment:
Not sure

```suggestion
  const lastInterpolatedPropsRef = React.useRef(skip ? props : initialProps ?? props);
```

Reply:
I don't think need this because a zero-duration transition is created when `skip` is true.

Nevertheless, I added a test covering this case to ensure we don't regress in the future.